src/stores/notifications/RoomNotificationStateStore.ts

```diff
@@ -108,6 +101,30 @@ export class RoomNotificationStateStore extends AsyncStoreWithClient<IState> {
         return RoomNotificationStateStore.internalInstance;
     }
 
+    private onSync = (state: SyncState, prevState?: SyncState, data?: ISyncStateData) => {
+        // Only count visible rooms to not torment the user with notification counts in rooms they can't see.
+        // This will include highlights from the previous version of the room internally
+        const globalState = new SummarizedNotificationState();
+        for (const room of this.matrixClient.getVisibleRooms()) {
+            if (VisibilityProvider.instance.isRoomVisible(room)) {
```

Comment:
I presume `getVisibleRooms` and `VisibilityProvider` check different things?

Reply:
Yes, `getVisibleRooms` is the js-sdk's interpretations which just filters out rooms which were upgraded.
VisibilityProvider is for react-sdk and element-web specifics, like hiding special Voip virtual rooms and whatnot